src/Sentry/Platforms/Native/CFunctions.cs

```diff
@@ -149,6 +239,17 @@ internal static void SetValueIfNotNull(sentry_value_t obj, string key, double? v
     internal static extern void sentry_remove_extra(string key);
 
     internal static Dictionary<long, DebugImage> LoadDebugImages(IDiagnosticLogger? logger)
+    {
+        // It only makes sense to load them once because they're cached on the native side anyway. We could force
+        // native to reload the list by calling sentry_clear_modulecache() when a dynamic library is loaded, but
+        // there's currently no way for us to know when that should happen.
+        DebugImages ??= LoadDebugImagesOnce(logger);
```

Comment:
We should document this. And if this becomes a problem maybe we hook into the .NET assembly loading and invalidate any time that happens (as it could reference new native libs)?

Reply:
https://github.com/getsentry/sentry-dotnet/issues/2922